Updates to netcdf-cxx4 tool

isabels opened the pull request.

isabels committed:
First try at a netcdf-cxx4 tool that works on linux/mac/windows.

isabels committed:
Add NETCDFCXX4DIR variable if pkg-config doesn't find it.

isabels committed:
Use LD_LIBRARY_PATH, if present, when running ldd.

isabels committed:
Add another libqxcb dependency to deploy_linux tool for qt5

garyjg commented:
It is unusual for a tool to create a variable only if pkg-config fails.  It means the developer cannot use the variable at all if pkg-config happens to succeed, which at the least might make testing difficult and also is probably not what someone expects.  If the variable is definitely not useful on OSX, then I guess it might make sense to not to create there, but if there's a chance it could be useful, then I suggest always creating the `NETCDFCXX4DIR` variable, with a default of None.  The None default implies use the platform-specific default: no change on Darwin, pkg-config if it works, and otherwise `/opt/local`.  Then a developer can set it explicitly anywhere, and if set it will be used as is, regardless of platform or pkg-config state.  Otherwise the default works everywhere, and there's no confusion about when the variable should or shouldn't exist.  Thoughts?

isabels committed:
Restructure to always create the NETCDFCXX4DIR variable

isabels commented:
Now that it looks like there's a package for netcdf-cxx4 on alma9, there's no current need for the NETCDFCXX4DIR variable. Should I get rid of it, or should we keep in case it's useful later?

garyjg committed:
find netcdf-cxx4 installed with cmake on fedora

since the netcdf-cxx4 library when built with cmake from source ends up with a
different name, go ahead and search for both names when the NETCDFCXX4DIR
variable is set, and in both subdirectory names, lib and lib64.  as before,
use the default library name for the platform unless pkg-config works.

garyjg commented:
I think the variable can still be useful, in particular I've been using it to test against the latest netcdf-cxx4 built from source.  So I've left it in and expanded the search to find "lib64/libnetcdf-cxx4.so" on Fedora.  Hopefully, this works the same as before when NETCDFCXX4DIR is not set, including on MacOS and Windows.
